pkg/agent/billing/billing.go

```diff
@@ -87,17 +93,34 @@ func RunBillingMetricsCollector(
 	store VMStoreForNode,
 	metrics PromMetrics,
 ) {
+	logger := parentLogger.Named("billing")
+
 	var clients []clientInfo
 
 	if c := conf.Clients.HTTP; c != nil {
 		clients = append(clients, clientInfo{
-			client: billing.NewClient(c.URL, http.DefaultClient),
+			client: billing.NewHTTPClient(c.URL, http.DefaultClient),
 			name:   "http",
 			config: c.BaseClientConfig,
 		})
 	}
-
-	logger := parentLogger.Named("billing")
+	if c := conf.Clients.S3; c != nil {
+		client, err := billing.NewS3Client(c.S3ClientConfig, time.Now)
+		if err != nil {
+			logger.Panic("Failed to create S3 client", zap.Error(err))
```

Comment:
This panic makes me a little nervous — I'm not sure how a panic here would be handled, given that it's a separate goroutine. I _think_ it'd take down the entire process?

What do you think about moving the `go RunBillingMetricsCollector` from `pkg/agent/entrypoint.go` into here, with a separate `StartBillingMetricsCollector()` that synchronously sets up the clients, so it can return an error if that fails?

Reply:
panics are orthogonal to the errgroup stuff right? Either way, we can discuss next week